src/Bundle/DependencyInjection/SyliusGridExtension.php

```diff
@@ -60,8 +62,23 @@ public function load(array $configs, ContainerBuilder $container): void
             ->addTag('sylius.grid')
         ;
 
+        $container->registerAttributeForAutoconfiguration(
+            AsFilter::class,
+            static function (ChildDefinition $definition, AsFilter $attribute, \Reflector $reflector): void {
+                // Helps to avoid issues with psalm
+                if (!$reflector instanceof \ReflectionClass) {
+                    return;
+                }
+
+                $definition->addTag(AsFilter::SERVICE_TAG, [
+                    'type' => $attribute->type ?? $reflector->getName(),
+                    'form_type' => $attribute->formType,
+                ]);
+            },
+        );
+
         $container->registerForAutoconfiguration(FilterInterface::class)
-            ->addTag('sylius.grid_filter')
+            ->addTag('sylius.legacy_grid_filter')
```

Comment:
We now tagged the Attribute instead, but we need to identify them for ConfigurableFilterInterface which contains static methods for the type and the form type.